tests: posix: Function call return values need to be validated.

sritej20 opened the pull request:
In this fix we are asserting the return value of pthread_attr_init.

Resolves #7084
Coverity-CID: 185280

Signed-off-by: Sritej Kanakadandi Venkata Rama <sritej.kvr@gmail.com>

sritej20 commented:
@nashif  I needed to add the newline to satisfy the checkpatch.pl requirement of 80 chars per line.

sritej20 commented:
@andrewboie @nashif @lpereira  Appreciate another final review to get the PR in.

andrewboie commented on tests/posix/pthread_rwlock/src/posix_rwlock.c:
```diff
@@ -80,7 +80,8 @@ static void test_rw_lock(void)
 	for (i = 0; i < N_THR; i++) {
-		pthread_attr_init(&attr[i]);
+		zassert_equal(pthread_attr_init(&attr[i]), 0,
+								"Unable to create pthread object attrib\n");
```
this should be aligned with the opening parenthesis of zassert_equal(), not way out over here

sritej20 replied:
Updated the patch . Please review .

sritej20 commented:
@andrewboie Updated the patch

sritej20 commented:
@andrewboie i guess we need a approval from you before the changes can be merged.

nashif commented:
> @nashif I needed to add the newline to satisfy the checkpatch.pl requirement of 80 chars per line.

not sure what you mean here, I am referring to "\n" at the end of the message.

sritej20 committed:
tests: posix: Function call return values need to be validated.

In this fix we are asserting the return value of pthread_attr_init.

Resolves #7084
Coverity-CID: 185280

Signed-off-by: Sritej Kanakadandi Venkata Rama <sritej.kvr@gmail.com>

sritej20 commented:
@nashif  My bad , Fixed that as well .
